src/Outsystems.SetupTools/Functions/Set-OSServerPerformanceTunning.ps1

```diff
@@ -52,165 +138,215 @@ function Set-OSServerPerformanceTunning
-            WriteNonTerminalError -Message "Error configuring IIS upload size limits"
+            # Configure .NET
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring .NET upload size limits and execution timeout (maxRequestLength = $MaxRequestLength, executionTimeout = $($OSPerfTuningExecutionTimeout.TotalSeconds) seconds)"
+            try
+            {
+                SetDotNetLimits -UploadLimit $MaxRequestLength -ExecutionTimeout $OSPerfTuningExecutionTimeout
+            }
+            catch
+            {
+                LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error configuring .NET settings"
+                WriteNonTerminalError -Message "Error configuring .NET settings"
 
-            return
+                return
+            }
         }
 
-        # Configure IIS for unlimited connections. (Default Web Site) - This should not be needed cause IIS defaults to maximum.
-        LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring IIS for unlimited connections (MaxConnections = 4294967295)"
-        try
-        {
-            SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost" -Name "sections['webLimits'].OverrideModeDefault" -Value "Allow"
-            SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost" -Name "sections['webLimits'].allowDefinition" -Value "Everywhere"
-            SetWebConfigurationProperty -PSPath "IIS:\" -Filter "system.applicationHost/sites/site[@name='Default Web Site']" -Name "Limits" -Value @{MaxConnections = 4294967295}
-        }
-        catch
+        if (-not $SkipIISUploadSizeLimitsConfig)
         {
-            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Exception $_.Exception -Stream 3 -Message "Error configuring IIS for unlimited connections"
-            WriteNonTerminalError -Message "Error configuring IIS for unlimited connections"
+            $MaxAllowedContentLength = $AdvancedConfigurations.Sections.IISUploadSizeLimitsConfig.NewMaxAllowedContentLength
 
-            return
-        }
+            if (-not $MaxAllowedContentLength)
+            {
+                $MaxAllowedContentLength = $OSPerfTuningMaxAllowedContentLength
+            }
 
-        # Configure IIS Application Pools
-        $DefaultWebSiteApps = $(Get-WebApplication -Site "Default Web Site").Path
+            # Configure IIS requests limits (Server Level)
+            LogMessage -Function $($MyInvocation.Mycommand) -Phase 1 -Stream 0 -Message "Configuring IIS upload size limits (maxAllowedContentLength = $MaxAllowedContentLength)"
+            try
+            {
+                SetWebConfigurationProperty -PSPath "MACHINE/WEBROOT/APPHOST" -Filter "system.webServer/security/requestFiltering/requestLimits" -Name "maxAllowedContentLength" -Value $MaxAllowedContentLength
```

Comment:
We should move this to a constant in src/Outsystems.SetupTools/Lib/Constants.ps1

Note: This maps to the registry pathin src/Outsystems.SetupTools/Functions/Get-OSPerfTuningStatus.ps1